src/routes/boot.ts

```diff
@@ -328,6 +331,48 @@ const getUser = (con: DataSource, userId: string): Promise<User> =>
     ],
   });
 
+const getMarketingCta = async (con: DataSource, userId: string) => {
+  let marketingCta: MarketingCta | null = null;
+  const redisKey = generateStorageKey(
+    StorageTopic.Boot,
+    'marketing_cta',
+    userId,
+  );
+  const rawRedisValue = await getRedisObject(redisKey);
+
+  if (rawRedisValue === RedisMagicValues.SLEEPING) {
+    // TODO: remove log or replace with proper logging
+    console.log('sleeping, not fetching from db');
+    return null;
+  } else {
+    marketingCta = JSON.parse(rawRedisValue);
+  }
+
+  if (!marketingCta) {
+    // TODO: remove log or replace with proper logging
+    console.log('fetching from db');
+    const userMarketingCta = await con.getRepository(UserMarketingCta).findOne({
+      where: { userId, readAt: IsNull() },
+      order: { createdAt: 'ASC' },
+      relations: ['marketingCta'],
+    });
+
+    if (userMarketingCta) {
+      marketingCta = userMarketingCta.marketingCta;
+      await setRedisObject(redisKey, JSON.stringify(marketingCta));
+    } else {
+      marketingCta = null;
+      // TODO: replace 10 with ONE_HOUR_IN_SECONDS
+      await setRedisObjectWithExpiry(redisKey, RedisMagicValues.SLEEPING, 10);
```

Comment:
Is 1h enough? Or maybe do a couple of hours? And when we do the customer.io integration, this redis key will be written over anyways, so don't think it hurts to have a long TTL 🤔

Reply:
Could probably even be indefinite right?
Since whatever will set it, will simply overwrite the value 👀

Reply:
Set it to infinity

Reply:
let's not set TTL for infinity on redis. We have to remember users are churning. Worst case we just query again the DB in real time and filling the cache

Reply:
How long should we set TTL to? 24 hours? More?

Reply:
Whatever works for you. Can be even days if you want. Just something